<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from drone.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="drone">
  <material name="frame_material">
    <color rgba="1 0.2 0.2 1"/>
  </material>
  <material name="arm_material">
    <color rgba="0.8 0.8 0.8 1"/>
  </material>
  <material name="propeller_material">
    <color rgba="0 0 0 0.6"/>
  </material>
  <link name="frame">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.1"/>
      </geometry>
      <material name="frame_material">
        <color rgba="0.8 0.8 0.8 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.1"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.88"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.002383332380000001" ixy="0.0" ixz="0.0" iyy="0.002383332380000001" iyz="0.0" izz="0.0044"/>
    </inertial>
  </link>
  <gazebo reference="frame">
    <material>Gazebo/Orange</material>
  </gazebo>
  <link name="arm1">
    <visual>
      <origin rpy="0 1.5707963267948966 0.7853981633974483" xyz="0.12374368670764582 0.1237436867076458 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
      <material name="arm_material"/>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 0.7853981633974483" xyz="0.12374368670764582 0.1237436867076458 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 1.5707963267948966 0.7853981633974483s" xyz="0.12374368670764582 0.1237436867076458 0.015000000000000001"/>
      <inertia ixx="1.8999992400000002e-05" ixy="0.0" ixz="0.0" iyy="1.8999992400000002e-05" iyz="0.0" izz="5.000000000000001e-07"/>
    </inertial>
  </link>
  <joint name="frame_arm1" type="fixed">
    <parent link="frame"/>
    <child link="arm1"/>
  </joint>
  <gazebo reference="arm1">
    <material>Gazebo/Grey</material>
  </gazebo>
  <link name="arm2">
    <visual>
      <origin rpy="0 1.5707963267948966 2.356194490192345" xyz="-0.1237436867076458 0.12374368670764582 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
      <material name="arm_material"/>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 2.356194490192345" xyz="-0.1237436867076458 0.12374368670764582 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 1.5707963267948966 2.356194490192345s" xyz="-0.1237436867076458 0.12374368670764582 0.015000000000000001"/>
      <inertia ixx="1.8999992400000002e-05" ixy="0.0" ixz="0.0" iyy="1.8999992400000002e-05" iyz="0.0" izz="5.000000000000001e-07"/>
    </inertial>
  </link>
  <joint name="frame_arm2" type="fixed">
    <parent link="frame"/>
    <child link="arm2"/>
  </joint>
  <gazebo reference="arm2">
    <material>Gazebo/Grey</material>
  </gazebo>
  <link name="arm3">
    <visual>
      <origin rpy="0 1.5707963267948966 3.9269908169872414" xyz="-0.12374368670764584 -0.1237436867076458 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
      <material name="arm_material"/>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 3.9269908169872414" xyz="-0.12374368670764584 -0.1237436867076458 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 1.5707963267948966 3.9269908169872414s" xyz="-0.12374368670764584 -0.1237436867076458 0.015000000000000001"/>
      <inertia ixx="1.8999992400000002e-05" ixy="0.0" ixz="0.0" iyy="1.8999992400000002e-05" iyz="0.0" izz="5.000000000000001e-07"/>
    </inertial>
  </link>
  <joint name="frame_arm3" type="fixed">
    <parent link="frame"/>
    <child link="arm3"/>
  </joint>
  <gazebo reference="arm3">
    <material>Gazebo/Grey</material>
  </gazebo>
  <link name="arm4">
    <visual>
      <origin rpy="0 1.5707963267948966 5.497787143782138" xyz="0.12374368670764578 -0.12374368670764584 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
      <material name="arm_material"/>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 5.497787143782138" xyz="0.12374368670764578 -0.12374368670764584 0.015000000000000001"/>
      <geometry>
        <cylinder length="0.15" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 1.5707963267948966 5.497787143782138s" xyz="0.12374368670764578 -0.12374368670764584 0.015000000000000001"/>
      <inertia ixx="1.8999992400000002e-05" ixy="0.0" ixz="0.0" iyy="1.8999992400000002e-05" iyz="0.0" izz="5.000000000000001e-07"/>
    </inertial>
  </link>
  <joint name="frame_arm4" type="fixed">
    <parent link="frame"/>
    <child link="arm4"/>
  </joint>
  <gazebo reference="arm4">
    <material>Gazebo/Grey</material>
  </gazebo>
  <link name="propeller1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
      <material name="propeller_material"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="2.5083323300000005e-05" ixy="0.0" ixz="0.0" iyy="2.5083323300000005e-05" iyz="0.0" izz="5e-05"/>
    </inertial>
  </link>
  <joint name="arm1_propeller1" type="revolute">
    <parent link="arm1"/>
    <child link="propeller1"/>
    <origin rpy="0 0 0" xyz="0.1767766952966369 0.17677669529663687 0.04"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <gazebo reference="propeller1">
    <material>Gazebo/RedTransparent</material>
  </gazebo>
  <link name="propeller2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
      <material name="propeller_material"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="2.5083323300000005e-05" ixy="0.0" ixz="0.0" iyy="2.5083323300000005e-05" iyz="0.0" izz="5e-05"/>
    </inertial>
  </link>
  <joint name="arm2_propeller2" type="revolute">
    <parent link="arm2"/>
    <child link="propeller2"/>
    <origin rpy="0 0 0" xyz="-0.17677669529663687 0.1767766952966369 0.04"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <gazebo reference="propeller2">
    <material>Gazebo/BlueTransparent</material>
  </gazebo>
  <link name="propeller3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
      <material name="propeller_material"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="2.5083323300000005e-05" ixy="0.0" ixz="0.0" iyy="2.5083323300000005e-05" iyz="0.0" izz="5e-05"/>
    </inertial>
  </link>
  <joint name="arm3_propeller3" type="revolute">
    <parent link="arm3"/>
    <child link="propeller3"/>
    <origin rpy="0 0 0" xyz="-0.17677669529663692 -0.17677669529663687 0.04"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <gazebo reference="propeller3">
    <material>Gazebo/BlueTransparent</material>
  </gazebo>
  <link name="propeller4">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
      <material name="propeller_material"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.1"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="2.5083323300000005e-05" ixy="0.0" ixz="0.0" iyy="2.5083323300000005e-05" iyz="0.0" izz="5e-05"/>
    </inertial>
  </link>
  <joint name="arm4_propeller4" type="revolute">
    <parent link="arm4"/>
    <child link="propeller4"/>
    <origin rpy="0 0 0" xyz="0.17677669529663684 -0.17677669529663692 0.04"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="0" upper="0" velocity="0"/>
  </joint>
  <gazebo reference="propeller4">
    <material>Gazebo/RedTransparent</material>
  </gazebo>
</robot>

